src/components/menu/styles.ts

```diff
@@ -18,3 +18,55 @@ export const ElMenuList = styled.div`
   background-color: var(--fill-white);
   box-shadow: 0px 4px 16px 0px #222b3329;
 `
+export const ElMenuItem = styled.li`
+  a,
+  button,
+  > div {
+    cursor: pointer;
+    display: flex;
+    align-items: center;
+    justify-content: start;
+    height: var(--size-7);
+    padding: var(--spacing-none) var(--spacing-4);
+    gap: var(--spacing-4);
+    font-family: var(--font-family);
+    font-size: var(--font-size-sm);
+    font-weight: 400;
+    line-height: var(--line-height-sm);
+    letter-spacing: var(--letter-spacing-sm);
+    color: var(--text-primary);
+    &:hover {
+      color: var(--text-white);
+      background: var(--fill-action-dark);
+    }
+  }
+
+  /* customize html default button */
+  button {
+    background-color: transparent;
+    border: none;
+    border-radius: 0;
+  }
```

Comment:
>  I'm really uncertain about the approach being taken here, specifically around the expectation for consumers to place a button inside a menu item. Here are my concerns/questions

for the context, I think that initially this menu will be used for app bar / navigation purpose

> Will this scale well? I've already identified <div role="button">, but what about other roles/variations on this same challenge?

yes, at the moment the styling only cover those 3 element defined in the style, but I think once other variants joining, which I think will also using div instead of input, its still fine 🤔 

> We want the whole menu item to be the hit area for pointer events, but it's not clear to me if this is currently the case given the elements that will have the click handlers are descendants of the menu item.

the custom handler will still be triggered (as long as the handler doesn't invoke `event.stopPropagation()`

> (Futurism) If we ever need menu items that look like an Element's button, these styles targeting any descendant <button> or <a> elements may get in the way (not 100% sure on this, as I guess it'll depend on rule specificity 🤔).

yes, I think that possible but I'm not sure if that is should be covered in here

Reply:
@nurm717123 I've been thinking more about the DOM structure within the menus and here's what I would propose we aim for in this PR:

1. No `<ul>` or `<li>` elements;
2. `Menu.List` renders to a `<div role="menu">`;
3. `Menu.Group` renders to a `<div role="group">`;
4. The group label renders as a `<div>`, maybe with `aria-hidden` (not sure, can always add that in future);
5. `Menu.Item` renders to a `<a role="menuitem">` or `<button role="menuitem">` depending on whether an `href` is provided (we're doing this for the Button work that is currently in PR and I think the same approach could work here); and,
6. We don't support or encourage nested interactive elements like `<button>` or `<a>` etc. Rather, the menu item is the thing that should be interactive.

These decisions would mean the DOM structure for a CSS only consumer's menu list might look something like the following snippets...

Simple:

```html
<div role="menu">
  <a role="menuitem">Link item</button>
  <button role="menuitem">Button item</button>
</div>
```

With a group:

```html
<div role="menu">
  <div role="group">
    <div>Some group</div>
    <a role="menuitem">Link item</button>
    <button role="menuitem">Button item</button>
  </div>
</div>
```

Some other comments on why I'm proposing this:
- From what I can tell from the ARIA docs, and looking at what other design systems have done, the element given the `menuitem` ARIA role is the element that should be interactive (not one of it's children);
- As such, it feels "safer" to me, if we are rendering `<a>` and `<button>` elements _as_ the menu items, not their children;
- This is just my proposal; there could be things I'm missing, so please let me know if you think there's problems with the above 🙏

Reply:
thanks for the example, as I've already not use `ul` and `li` I think I just need to have a few amendment after this
1. move `role=menu` from popover to the list, I think its better because this way we can move out the popover if we want
2. use `a` or `button` depends on the `href` props, totally agree since it was my 1st proposal to achieve this (not in this repo) but the cons is the use might still put a interactive elements under it like button, if they are not aware of it, that's the reason, but I think that's not our fault.
3. the group with `aria-hidden`, I think we don't need this as we don't render the component at all when it is hidden, unless we use `display: none` approach.